app/src/main/java/com/microsoft/graph/snippets/application/SnippetApp.java

```diff
@@ -6,34 +6,45 @@
 
 import android.app.Application;
 import android.content.Context;
-
-import com.microsoft.graph.core.DefaultClientConfig;
-import com.microsoft.graph.core.IClientConfig;
-import com.microsoft.graph.extensions.GraphServiceClient;
-import com.microsoft.graph.extensions.IGraphServiceClient;
+import com.microsoft.graph.authentication.MSALAuthenticationProvider;
+import com.microsoft.graph.requests.extensions.GraphServiceClient;
+import com.microsoft.graph.models.extensions.IGraphServiceClient;
 import com.microsoft.graph.snippets.AuthenticationManager;
-
+import com.microsoft.graph.snippets.ServiceConstants;
+import com.microsoft.graph.snippets.SignInActivity;
 
 public class SnippetApp extends Application {
     private static SnippetApp sSnippetApp;
+    private static SignInActivity signInActivity;
     private AuthenticationManager mAuthenticationManager;
 
     public static SnippetApp getApp() {
         return sSnippetApp;
     }
+    public static SignInActivity getAppActivity() {return signInActivity;}
 
     @Override
     public void onCreate() {
         super.onCreate();
         sSnippetApp = this;
+        signInActivity = new SignInActivity();
         mAuthenticationManager = AuthenticationManager.getInstance();
     }
 
     public IGraphServiceClient getGraphServiceClient() {
-        IClientConfig clientConfig = DefaultClientConfig.createWithAuthenticationProvider(
-                mAuthenticationManager
-        );
-        return new GraphServiceClient.Builder().fromConfig(clientConfig).buildClient();
+
+        MSALAuthenticationProvider msalAuthenticationProvider = new MSALAuthenticationProvider(
```

Comment:
You can store msalAuthenticationProvider in a member variable and initilize it only once when its null. Same with graphClient.

Reply:
Updated